samples/ClientSample/Program.cs

```diff
@@ -45,56 +47,22 @@ public class Program
                         cts.Cancel();
                     };
 
-                    // Ready to start the loops
-                    var receive = StartReceiving(loggerFactory.CreateLogger("ReceiveLoop"), connection, cts.Token);
-                    var send = StartSending(loggerFactory.CreateLogger("SendLoop"), connection, cts.Token);
-
-                    await Task.WhenAll(receive, send);
-                }
-            }
-        }
-
-        private static async Task StartSending(ILogger logger, Connection connection, CancellationToken cancellationToken)
-        {
-            logger.LogInformation("Send loop starting");
-            while (!cancellationToken.IsCancellationRequested)
-            {
-                var line = Console.ReadLine();
-                logger.LogInformation("Sending: {0}", line);
-
-                await connection.Output.WriteAsync(Encoding.UTF8.GetBytes(line));
-            }
-            logger.LogInformation("Send loop terminated");
-        }
-
-        private static async Task StartReceiving(ILogger logger, Connection connection, CancellationToken cancellationToken)
-        {
-            logger.LogInformation("Receive loop starting");
-            using (cancellationToken.Register(() => connection.Input.Complete()))
-            {
-                while (!cancellationToken.IsCancellationRequested)
-                {
-                    var result = await connection.Input.ReadAsync();
-                    var buffer = result.Buffer;
-                    try
+                    // Set up handler
+                    connection.On("Send", new[] { typeof(string) }, a =>
```

Comment:
We would need to add [generic extensions](https://github.com/SignalR/SignalR/blob/dev/src/Microsoft.AspNet.SignalR.Client/HubProxyExtensions.cs#L122) later...